src/order/bounds.lean

```diff
@@ -1,378 +1,609 @@
+
+lemma union_upper_bounds_subset_upper_bounds_inter :
+  upper_bounds s ∪ upper_bounds t ⊆ upper_bounds (s ∩ t) :=
+union_subset
+  (upper_bounds_mono_set $ inter_subset_left _ _)
+  (upper_bounds_mono_set $ inter_subset_right _ _)
+
+lemma union_lower_bounds_subset_lower_bounds_inter :
+  lower_bounds s ∪ lower_bounds t ⊆ lower_bounds (s ∩ t) :=
+@union_upper_bounds_subset_upper_bounds_inter (order_dual α) _ s t
+
+lemma is_least_union_iff {a : α} {s t : set α} :
+  is_least (s ∪ t) a ↔ (is_least s a ∧ a ∈ lower_bounds t ∨ a ∈ lower_bounds s ∧ is_least t a) :=
+by simp [is_least, lower_bounds_union, or_and_distrib_right, and_comm (a ∈ t), and_assoc]
+
+lemma is_greatest_union_iff :
+  is_greatest (s ∪ t) a ↔ (is_greatest s a ∧ a ∈ upper_bounds t ∨
+    a ∈ upper_bounds s ∧ is_greatest t a) :=
+@is_least_union_iff (order_dual α) _ a s t
+
+/-- If `s` is bounded, then so is `s ∩ t` -/
+lemma bdd_above.inter_of_left (h : bdd_above s) : bdd_above (s ∩ t) :=
+h.mono $ inter_subset_left s t
 
-section lattice
+/-- If `t` is bounded, then so is `s ∩ t` -/
+lemma bdd_above.inter_of_right (h : bdd_above t) : bdd_above (s ∩ t) :=
+h.mono $ inter_subset_right s t
 
-lemma is_glb_empty [order_top α] : is_glb ∅ (⊤:α) :=
-by simp [is_glb, is_greatest, lower_bounds, upper_bounds]
+/-- If `s` is bounded, then so is `s ∩ t` -/
+lemma bdd_below.inter_of_left (h : bdd_below s) : bdd_below (s ∩ t) :=
+h.mono $ inter_subset_left s t
 
-lemma is_lub_empty [order_bot α] : is_lub ∅ (⊥:α) :=
-by simp [is_lub, is_least, lower_bounds, upper_bounds]
+/-- If `t` is bounded, then so is `s ∩ t` -/
+lemma bdd_below.inter_of_right (h : bdd_below t) : bdd_below (s ∩ t) :=
+h.mono $ inter_subset_right s t
 
-lemma is_lub_union_sup [semilattice_sup α] (hs : is_lub s a₁) (ht : is_lub t a₂) :
-  is_lub (s ∪ t) (a₁ ⊔ a₂) :=
+/-- If `s` and `t` are bounded above sets in a `semilattice_sup`, then so is `s ∪ t`. -/
+lemma bdd_above.union [semilattice_sup γ] {s t : set γ} :
+  bdd_above s → bdd_above t → bdd_above (s ∪ t) :=
+begin
+  rintros ⟨bs, hs⟩ ⟨bt, ht⟩,
+  use bs ⊔ bt,
+  rw upper_bounds_union,
+  exact ⟨upper_bounds_mono_mem le_sup_left hs,
+    upper_bounds_mono_mem le_sup_right ht⟩
+end
+
+/-- The union of two sets is bounded above if and only if each of the sets is. -/
+lemma bdd_above_union [semilattice_sup γ] {s t : set γ} :
+  bdd_above (s ∪ t) ↔ bdd_above s ∧ bdd_above t :=
+⟨λ h, ⟨h.mono $ subset_union_left s t, h.mono $ subset_union_right s t⟩,
+  λ h, h.1.union h.2⟩
+
+lemma bdd_below.union [semilattice_inf γ] {s t : set γ} :
+  bdd_below s → bdd_below t → bdd_below (s ∪ t) :=
+@bdd_above.union (order_dual γ) _ s t
+
+/--The union of two sets is bounded above if and only if each of the sets is.-/
+lemma bdd_below_union [semilattice_inf γ] {s t : set γ} :
+  bdd_below (s ∪ t) ↔ bdd_below s ∧ bdd_below t :=
+@bdd_above_union (order_dual γ) _ s t
+
+/-- If `a` is the least upper boundary of `s` and `b` is the least upper boundary of `t`,
```

Comment:
You have `boundary` instead of `bound` in this docstring, and also below.

Reply:
In Russian these two words are even more similar...